ocaml/parsing/pprintast.ml

```diff
@@ -1575,7 +1575,7 @@ and payload ctxt f = function
       pp f "?@ "; pattern ctxt f x;
       pp f " when "; expression ctxt f e
 
-and pp_print_pexp_function ctxt sep f x =
+and pp_print_pexp_newtype ctxt sep f x =
```

Comment:
To make sure I'm understanding the correctly: I assume there was some sort of jane-syntax related issue when an `x` with description `Pexp_function` is passed here? What exactly was the issue?

Reply:
oh, sorry, thought I wrote this down. the issue is in ppxlib's round trip check: it reparses the output of pprintast and complains if it differs from the original parsetree. It's nice for `Pexp_newtype (a, Pexp_function (...))` to be printed differently than `Pexp_function ([ Pparam_newtype a; ... ], ...)` for that reason.